cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/model/keys/provider/ProviderKeyPair.java

```diff
@@ -9,11 +15,17 @@ public ProviderKeyPair(ProviderSecretKey sk, ProviderPublicKey pk) {
         this.pk = pk;
     }
 
-    public ProviderPublicKey getPk() {
-        return this.pk;
+    /**
+     * constructor for construction of object by deserialization
+     * @param repr serialized representation
+     */
+    public ProviderKeyPair(Representation repr)
+    {
+        new ReprUtil(this).deserialize(repr); // side effect reflection magic used to restore fields
     }
 
-    public ProviderSecretKey getSk() {
-        return this.sk;
+    public Representation getRepresentation()
+    {
+        return ReprUtil.serialize(this);
```

Comment:
Add unit tests to check whether the serialization and deserialization actually works :)

Reply:
Done